crates/torii/core/src/processors/event_message.rs

```diff
@@ -0,0 +1,57 @@
+use anyhow::{Error, Result};
+use async_trait::async_trait;
+use dojo_world::contracts::model::ModelReader;
+use dojo_world::contracts::world::WorldContractReader;
+use starknet::core::types::{BlockWithTxs, Event, TransactionReceipt};
+use starknet::providers::Provider;
+use tracing::info;
+
+use super::EventProcessor;
+use crate::processors::MODEL_INDEX;
+use crate::sql::Sql;
+
+#[derive(Default)]
+pub struct EventMessageProcessor;
+
+#[async_trait]
+impl<P> EventProcessor<P> for EventMessageProcessor
+where
+    P: Provider + Send + Sync,
+{
+    fn event_key(&self) -> String {
+        "".to_string()
+    }
+
+    fn validate(&self, _event: &Event) -> bool {
+        true
```

Comment:
We can check here that the `keys` len must be at least 3:
1. Event selector
2. Models must have at least 1 key
3. System address

Reply:
Oh right makes sense, nice catch